skbio/diversity/__init__.py

```diff
@@ -16,6 +16,233 @@
    alpha
    beta
 
+Functions
+---------
+
+.. autosummary::
+   :toctree: generated/
+
+    alpha_diversity
+    beta_diversity
+    get_alpha_diversity_metrics
+    get_beta_diversity_metrics
+
+Examples
+--------
+
+Create a table containing 7 OTUs and 6 samples:
+
+.. plot::
+   :context:
+
+>>> import numpy as np
+>>> data = [[23, 64, 14, 0, 0, 3, 1],
+...         [0, 3, 35, 42, 0, 12, 1],
+...         [0, 5, 5, 0, 40, 40, 0],
+...         [44, 35, 9, 0, 1, 0, 0],
+...         [0, 2, 8, 0, 35, 45, 1],
+...         [0, 0, 25, 35, 0, 19, 0]]
+>>> ids = list('ABCDEF')
+
+Compute observed OTUs for each sample:
+
+>>> from skbio.diversity import alpha_diversity
+>>> alpha_diversity('observed_otus', data, ids)
+A    5
+B    5
+C    4
+D    4
+E    5
+F    3
+dtype: int64
+
+Next we'll compute Faith's PD on the same samples. Since this is a phylogenetic
+diversity metric, we'll first need to create a tree and an ordered list of
+OTU identifiers:
+
+>>> from skbio import TreeNode
+>>> from io import StringIO
+>>> tree = TreeNode.read(StringIO(
+...                      '(((((OTU1:0.5,OTU2:0.5):0.5,OTU3:1.0):1.0):0.0,'
+...                      '(OTU4:0.75,(OTU5:0.5,(OTU6:0.5,OTU7:0.5):0.5):'
+...                      '0.5):1.25):0.0)root;'))
+>>> otu_ids = ['OTU1', 'OTU2', 'OTU3', 'OTU4', 'OTU5', 'OTU6', 'OTU7']
+
+Because ``faith_pd`` takes ``otu_ids`` and ``tree`` as additional parameters,
+those are required to be passed as ``kwargs`` to ``alpha_diversity``.
+
+>>> alpha_diversity('faith_pd', data, ids=ids, otu_ids=otu_ids, tree=tree)
+A    6.75
+B    7.00
+C    6.25
+D    5.75
+E    6.75
+F    5.50
+dtype: float64
+
+Note that we passed ``'faith_pd'`` as a string to ``alpha_diversity``. While we
+could have passed the function itself (i.e., ``metric=faith_pd``) passing it as
+a string results in an optmized verison of the function being used. Wherever
+possible, you should pass ``metric`` as a string which will result in an
+optimized version being used if available. Passing ``metric`` as a string may
+not be possible if you're passing a metric that scikit-bio doesn't know about,
+such as a custom one that you've developed.
+
+
+The value that you provide for to ``beta_diversity`` for ``metric`` can
+be either a string (e.g., "unweighted_unifrac") or a function
+(e.g., ``skbio.diversity.beta.unweighted_unifrac``). The metric should
+generally be passed as a string, as this often uses an optimized version
```

Comment:
this point about optimized version seems redundant as that was discussed already in the previous paragraph